src/Containers/MovieListPage/MovieListPage.jsx

```diff
@@ -1,77 +1,68 @@
-import { useState } from 'react';
 import { GenreSelect } from '../../Components/GenreSelect/GenreSelect';
-import { SearchForm } from '../../Components/SearchForm/SearchForm';
-import { genres, sortOptions } from '../../constants';
-import { MovieDetails } from '../../Components/MovieDetails/MovieDetails';
+import { formGenres, qenreParamName, queryParamName, sortParamName } from '../../constants';
 import { MovieTile } from '../../Components/MovieTile/MovieTile';
 import { SortControl } from '../../Components/SortControl/SortControl';
+import styles from './MovieListPage.module.css';
+import { useFetchMovies } from '../../customHooks/moviesHooks';
+import { Link, Outlet, useLocation, useSearchParams } from 'react-router-dom';
 import { Dialog } from '../../Components/Dialog/Dialog';
 import { MovieForm } from '../../Components/MovieForm/MovieForm';
-import styles from './MovieListPage.module.css';
-import { useFetchMovies } from '../../customHooks/movies';
+import { useState } from 'react';
 
 export function MovieListPage() {
-  const [selectedGenre, setSelectedGenre] = useState(genres[0]);
-  const [searchQuery, setSearchQuery] = useState('');
-  const [selectedSortOption, setSelectedSortOption] = useState(sortOptions[0].value);
-  const [selectedMovie, setSelectedMovie] = useState();
+  const [searchParams] = useSearchParams();
+  const location = useLocation();
+  const isMovieIdPath = /^\/\d+\/?$/.test(location.pathname);
```

Comment:
Maybe `useParams` will be more convenient to use:

```suggestion
  const { movieId } = useParams();
  const isMovieIdPath = !!movieId;
```

or `useMatch`:

```suggestion
  const isMovieIdPath = useMatch('/movie/:movieId');
```

?